### ISSUE TITLE
Use of dynamically named variables instead of dictionaries

### ISSUE THREAD

**Issue by filbranden** [vim-latex/vim-latex#172]
vim-latex uses dynamically named variables such as `Tex_Env_{name}`, `Tex_Com_{name}`, `Tex_ViewRule_{format}`, etc.

Some of these present shortcomings due to what characters are allowed in variable names. For instance, it's not really possible to use `Tex_Env_*` to set a custom environment for `align*`, since the `*` is not an allowed character in a variable name. It turns out this used to work in Vim 6, but hasn't since Vim 7 (released almost 14 years ago.) The specific example of `g:Tex_Env_{'align*'}` is in vim-latex's documentation [here](http://vim-latex.sourceforge.net/documentation/latex-suite.html#customizing-macros).

It seems most of this was developed before Vim had support for dicts... Since it does not (for over a decade), should we switch to using dicts `Tex_Env`, `Tex_Com`, `Tex_ViewRule`, etc. to configure those? (We could still query the legacy values for compatibility sake, but prefer the newer dicts and document those only.)

See discussion in [this Stack Exchange Q & A](https://vi.stackexchange.com/q/24467/18609) for more context.

**Comment by gerw** [comment]
Yes, this would be very helpful. Due to lack of time, I am not able to include this change in the near future. Pull requests are welcome ;)
